server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -566,6 +568,13 @@ final SearchContext createAndPutContext(ShardSearchRequest request) throws IOExc
 
     final SearchContext createContext(ShardSearchRequest request) throws IOException {
         final DefaultSearchContext context = createSearchContext(request, defaultSearchTimeout);
+        if (activeContexts.size() >= Node.MAX_SEARCH_CONTEXT_SETTING.get(settings)) {
+            throw new IllegalStateException(
```

Comment:
I think ElasticsearchException would allow to return a HTTP status code that indicates that the request should be retried later, this would be better than a `500/Internal Error`?